dbms/src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -90,11 +90,28 @@ BlockIO InterpreterDropQuery::executeToTable(String & database_name_, String & t
             /// If table was already dropped by anyone, an exception will be thrown
 
             auto table_lock = database_and_table.second->lockExclusively(context.getCurrentQueryId());
-            /// Delete table metadata and table itself from memory
 
+            const auto prev_metadata_name = database_and_table.first->getMetadataPath() + escapeForFileName(database_and_table.second->getTableName()) + ".sql";
```

Comment:
It's totally and absolutely wrong that metadata is manipulated here (it's internal detail of DatabaseOrdinary).